src/main/java/org/inferred/freebuilder/processor/Metadata.java

```diff
@@ -135,6 +135,12 @@ public ParameterizedType getBuilder() {
   /** Returns metadata about the properties of the type. */
   public abstract ImmutableList<Property> getProperties();
 
+  /** Returns super types implementing builder */
+  public abstract ImmutableSet<ParameterizedType> getSuperBuilderTypes();
+
+  /** Returns super type properties */
+  public abstract ImmutableMap<ParameterizedType, ImmutableList<Property>> getSuperTypeProperties();
```

Comment:
Also, generic builders and generic supertypes will be problematic. For generic builders, you'll have to be very careful you're putting the right type parameters in the right places (which you may already be doing, but I'd need to see tests). For generic supertypes, you'll need to figure out the correct wildcard parameterisation to put in.